plugins/ssr/ssrequality.ml

```diff
@@ -155,10 +166,12 @@ let newssrcongrtac arg ist =
     tclMATCH_GOAL (arrow, gl') (fun gl' -> [|fs gl' lhs;fs gl' rhs|])
     (fun lr ->
       let a = ssr_congr lr in
-      tclTHENLIST [ pf_typecheck a
-                  ; Proofview.V82.of_tactic (Tactics.apply a)
+      Tacticals.New.tclTHENLIST [ pf_typecheck a
+                  ; Tactics.apply a
                   ; congrtac (arg, mkRType) ist ])
-    (fun _ _ -> errorstrm Pp.(str"Conclusion is not an equality nor an arrow")))
+    (fun _ -> Tacticals.New.tclZEROMSG Pp.(str"Conclusion is not an equality nor an arrow"))
+    with e -> Proofview.tclZERO e (* FIXME *)
```

Comment:
This is the only fishy thing

Reply:
If I'm not mistaken the old code was letting the exception escape. Questions:
- who can raise it? (I can't find a candidate)
- this may make the monad backtrack, while I guess an escaping exception is fatal. Is this true?

Reply:
@ppedrot can you clarify (possibly in the comment)?

Reply:
There is some weird things going on w.r.t. exceptions in the current state of the code, but I don't remember whether this with handler was needed or if it was breaking things.

Stray exceptions are problematic in the tactic monad, because they do not magically become fatal. If they happen to be raised from withing a block like e.g. `Proofview.Goal.enter` they might be wrongfully caught. This is highly non local, so virtually impossible to track until we cleanup the whole implementation. I encountered some stray exceptions leading to unexpected failures, so I prefer to be as conservative as possible in the current code.